03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,338 @@
+---
+title: "Clustering Heatmap - RNA-Seq"
+author: "CCDL for ALSF"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+
+*Purpose*: This notebook illustrates one way that you can use harmonized data 
+from refine.bio in downstream analyses.
+
+TODO: This introduction section will be revamped when the intro to an example
+analysis (#122: https://github.com/AlexsLemonade/refinebio-examples/pull/122)
+gets more polished/determined.
+
+## 1) Install libraries
+This script uses the R package `DESeq2` for normalization and the R package
+`pheatmap` for clustering and creating a heatmap.
+<br/> <i> Citation:</i> Love, M.I., Huber, W., Anders, S. Moderated estimation of fold change and dispersion for RNA-Seq data with DESeq2 Genome Biology 15(12):550 (2014)
+<br/> <i> Citation:</i> Raivo Kolde (2019). pheatmap: Pretty Heatmaps. R package version 1.0.12. https://CRAN.R-project.org/package=pheatmap
+
+```{r Install and attach the pheatmap and DESeq2 libraries}
+if (!("pheatmap" %in% installed.packages())) {
+  # Install pheatmap
+  install.packages("pheatmap", update = FALSE)
+}
+
+if (!("DESeq2" %in% installed.packages())) {
+  # Install DESeq2
+  BiocManager::install("DESeq2", update = FALSE)
+}
+```
+
+Attach the `pheatmap` and `DESeq2` libraries:
+
+```{r}
+# Attach the `pheatmap` library
+library(pheatmap)
+
+# Attach the `DESeq2` library
+library(DESeq2)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Set the seed so our results are reproducible:
+set.seed(12345)
+```
+
+Create output folder.
+
+```{r}
+# Define the file path to the plots directory
+plots_dir <- "plots" # Replace with path to desired output plots directory
+
+# Create the plots folder if it doesn't exist
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+## 2) Import and set up data
+Data downloaded from refine.bio include a metadata [tab separated values ("tsv")]
+(https://en.wikipedia.org/wiki/Tab-separated_values) file and a data tsv file.
+This chunk of code will read the both tsv files and add them as data.frame
+objects to your environment.
+
+The data that we downloaded from refine.bio for this analysis is a dataset, with
+a total of 24 samples (obtained from 6 mice with 4 technical replicates for each
+mouse), containing RNA-sequencing results for Group 3 and 4 medulloblastoma
+orthotopic xenograft mice with and without digoxin treatment.
+
+You can download this dataset by going to the following link and clicking 
+`Download Now`: https://www.refine.bio/experiments/SRP150101/transcriptome-analysis-of-group-3-and-4-medulloblastoma-orthotopic-xenograft-mice-with-digoxin-treatment
```

Comment:
This part is actually not necessary here, because it will be covered when we add in the getting started section for this module (being workshopped over on #122 ).

Reply:
But you can keep it here for now I guess.

Reply:
In your [previous review comment](https://github.com/AlexsLemonade/refinebio-examples/pull/140#discussion_r456024079), you mentioned that there is a likelihood of someone skipping over this in the getting started section so it is good to have here as well.